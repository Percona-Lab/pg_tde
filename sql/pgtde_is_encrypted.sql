CREATE EXTENSION pg_tde;

SELECT  * FROM pg_tde_master_key_info();

SELECT pg_tde_add_key_provider_file('file-vault','/tmp/pg_tde_test_keyring.per');
SELECT pg_tde_set_master_key('test-db-master-key','file-vault');

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING pg_tde;

CREATE TABLE test_norm(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING heap;

SELECT amname FROM pg_class INNER JOIN pg_am ON pg_am.oid = pg_class.relam WHERE relname = 'test_enc';
SELECT amname FROM pg_class INNER JOIN pg_am ON pg_am.oid = pg_class.relam WHERE relname = 'test_norm';

SELECT pgtde_is_encrypted('test_enc');
SELECT pgtde_is_encrypted('test_norm');

SELECT  key_provider_id, key_provider_name, master_key_name
		FROM pg_tde_master_key_info();

DROP TABLE test_enc;
DROP TABLE test_norm;

DROP EXTENSION pg_tde;
