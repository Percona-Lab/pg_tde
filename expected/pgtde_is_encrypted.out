CREATE EXTENSION pg_tde;
SELECT  * FROM tde_master_key_info();
ERROR:  Master key does not exists for the database
HINT:  Use set_master_key interface to set the master key
SELECT pg_tde_add_key_provider_file('file-vault','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_key_provider_file 
------------------------------
                            1
(1 row)

SELECT pg_tde_set_master_key('test-db-master-key','file-vault');
 pg_tde_set_master_key 
-----------------------
 t
(1 row)

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING pg_tde;
CREATE TABLE test_norm(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING heap;
SELECT amname FROM pg_class INNER JOIN pg_am ON pg_am.oid = pg_class.relam WHERE relname = 'test_enc';
 amname 
--------
 pg_tde
(1 row)

SELECT amname FROM pg_class INNER JOIN pg_am ON pg_am.oid = pg_class.relam WHERE relname = 'test_norm';
 amname 
--------
 heap
(1 row)

SELECT pgtde_is_encrypted('test_enc');
 pgtde_is_encrypted 
--------------------
 t
(1 row)

SELECT pgtde_is_encrypted('test_norm');
 pgtde_is_encrypted 
--------------------
 f
(1 row)

SELECT  key_provider_id, key_provider_name, master_key_name
		FROM tde_master_key_info();
 key_provider_id | key_provider_name |  master_key_name   
-----------------+-------------------+--------------------
               1 | file-vault        | test-db-master-key
(1 row)

DROP TABLE test_enc;
DROP TABLE test_norm;
DROP EXTENSION pg_tde;
